src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/runes/SoulboundRune.java

```diff
@@ -59,7 +60,7 @@ public ItemDropHandler getItemHandler() {
         };
     }
 
-    private void activate(Player p, Item rune) {
+    private void activate(@Nonnull Player p, @Nonnull Item rune) {
```

Comment:
No need for Validate.notNull here?
Or is it checked upon in the methods which it calls later on?

Reply:
It's a private method anyway.
Only we can call it, so there isn't really any need for param validation since we have full control over what goes into this method.